metadata/src/utils/retain.rs

```diff
@@ -356,4 +428,34 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn issue_1659() {
+        let metadata_cache = load_metadata("../artifacts/regressions/1659.scale");
+
+        // Strip metadata to the pallets as described in the issue.
+        let mut stripped_metadata = metadata_cache.clone();
+        retain_metadata(
+            &mut stripped_metadata,
+            {
+                let set = "Balances,Timestamp,Contracts,ContractsEvm,System"
+                    .split(",")
+                    .collect::<BTreeSet<&str>>();
+                move |s| set.contains(&s)
+            },
```

Comment:
I'd simplify this to just `|s| ["Balances","Timestamp","Contracts","ContractsEVM","System"].contains(s)`; no need for the set etc really (and with so few items it's prob not even slower anyways, but being a test it's not a big deal)

Reply:
i written it like this only because repro command is a string and i was lazy to type